metricbeat/module/vsphere/network/network.go

```diff
@@ -0,0 +1,135 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/module/vsphere"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/mo"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each network is defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("vsphere", "network", New,
+		mb.WithHostParser(vsphere.HostParser),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet type defines all fields of the MetricSet.
+type MetricSet struct {
+	*vsphere.MetricSet
+}
+
+// New creates a new instance of the MetricSet.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	ms, err := vsphere.NewMetricSet(base)
+	if err != nil {
+		return nil, err
+	}
+	return &MetricSet{ms}, nil
+}
+
+type metricData struct {
+	assetsName assetNames
+}
+
+type assetNames struct {
+	outputVmNames   []string
+	outputHostNames []string
+}
+
+// Fetch method implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
More of a question here @kush-elastic. Could it be possible for the fetch method to hit a bottleneck depending on the size of the vSphere environment?

The fix that we should keep in mind should this ever happen would be to implement some sort of pagination using a batch size.

Reply:
We are only fetching the necessary information, so I believe we won't encounter this problem. However, I will still investigate pagination options.